rfc/halyard-lite.md

```diff
@@ -0,0 +1,508 @@
+*   Edit the YAML directly
+*   Run `hal …` commands to update the config
+
+YAML is ubiquitous in the Kubernetes world, and Kubernetes-native tools generally
+assume that users are comfortable writing YAML configuration files. As such, the
+value in providing a CLI that hides YAML from users is questionable, and may even
+contribute to the perception that Spinnaker (or at least its installation) is not
+Kubernetes-native.  The CLI also encourages an imperative workflow whereby
+configuration is achieved via a series hal commands.  Furthermore, a significant
+amount of the complexity and verbosity of Halyard’s codebase comes from the commands
+to edit the Halyard configuration.
+
+Based on these considerations, the CLI commands to edit the configuration will be
+removed.  As the [hal command reference](https://www.spinnaker.io/reference/halyard/commands/)
+is the best existing documentation on the available config parameters for Spinnaker,
+it will be replaced by extensive documentation of a well-typed Halyard configuration
+so that users can be confident directly editing this config.
+
+#### Generation and application of Kubernetes YAML
+
+As discussed above, Halyard’s functionality to generate and apply Kubernetes YAML
+duplicates functionality in existing tools and makes it incompatible with these tools.
+It also makes it impossible for users to customize their deployments without making
+code changes to Halyard to account for these customizations.
+
+For these reasons, Halyard will no longer handle generating and deploying Kubernetes
+YAML.  Instead, Halyard will focus on generating the service configs from the Halyard
+config.  We’ll provide a Helm chart and/or Kustomization that consumes these service
+configs and generates/applies YAML to the cluster.
+
+#### Input validation
+
+Halyard currently does fairly strict validation of input when using the CLI; in
+addition to validating the well-formedness of parameters, in some cases it also
+uses the code from the other services to attempt to perform operations using the
+parameters.  (For example, when configuring a Google Compute account, Halyard will
+try to make a test request to list instances during its validation step.)
+
+The value of this validation is not clear when Halyard is running on a different
+machine than the microservice in question.  In many cases, network policies or
+firewalls prevent the Halyard instance from communicating with external services that
+individual microservices do have access to.
+
+Halyard will reduce the scope of its validation efforts.  It will focus on
+validating the well-formedness of the Halyard config, but will no longer
+act as the other microservices to try to actually build instances of credentials
+and make requests using them.  Validating the well-formedness of the config will
+involve both:
+* Validating the structure of the config
+* Performing simple validation on the values supplied in the config, such as
```

Comment:
Halyard also performed some additional simple validation that doesn't fall into the well-formedness or auth* categories: 
```
+ Get current deployment
  Success
+ Get persistent store
  Success
Validation in default.persistentStorage:
- WARNING Your deployment will most likely fail until you configure
  and enable a persistent store.
```

Will simple checks like this still be in scope?

Reply:
I think checks like that would be in scope; I suppose that's a bit broader than just checking each individual value, but I could imagine things of the type "you must have at least one of X" configured would be in scope.